### PR TITLE
adds checkout redirect to nav menu

### LINKED ISSUE #244: Update the Buy page with link to Stripe page
Screenshot 2020-11-16 at 16 18 34

### PR DESCRIPTION
**Pull request by sssoja**
### What changes have you made?

- I've set up the code to allow for a redirect to Stripe checkout from the "Buy" item in the nav menu 
- To avoid having a massive list of allowed countries, I've moved the array into an exportable const called `allowedCountries`. I wasn't sure where to put it so I created a file called `constants.ts` in a lib folder 
- As we're using the same `handleClick` in the `NavMenu` as in the `BuyButton` component it seems overkill to repeat so much code in another sibling so I've created a helper called `redirectToCheckout`
- I'm just missing how to implement the helper at the component level which is why this PR has the block of code commented out in `NavMenu` and the helper isn't returning anything other than `{error}` on line 44 of `redirectToCheckout`.

Essentially I don't know if we need a helper, maybe we don't, but if we do I'd like to replace the block of code that's commented out with `redirectToCheckout`.

### Which issue(s) does this PR fix?

Fixes #244

### CODE REVIEW

**Comment by Jbarget** [comment]
fixed :) 

so the helper as you made it was a **React functional component**, helpers are generally **functions**.
The only thing that needs to be a helper is the `onClick` function rather than a whole component.

Good thinking to extract it out and keep things dry!

File: public/locales/es/translation.json [hunk 1]
```diff
@@ -37,8 +37,11 @@
...
-      "voltrova": "Lorem Ipsum es simplemente el texto de relleno de las imprentas y archivos de texto. Lorem Ipsum ha sido el texto de relleno estándar de las industrias desde el año 1500, cuando un impresor (N. del T. persona que se dedica a la imprenta) desconocido usó una galería de textos y los mezcló de tal manera que logró hacer un libro de textos especimen."
+  "advertising": {
+    "salazraki": "Lorem Ipsum es simplemente el texto de relleno de las imprentas y archivos de texto. Lorem Ipsum ha sido el texto de relleno estándar de las industrias desde el año 1500, cuando un impresor (N. del T. persona que se dedica a la imprenta) desconocido usó una galería de textos y los mezcló de tal manera que logró hacer un libro de textos especimen.",
+    "voltrova": "Lorem Ipsum es simplemente el texto de relleno de las imprentas y archivos de texto. Lorem Ipsum ha sido el texto de relleno estándar de las industrias desde el año 1500, cuando un impresor (N. del T. persona que se dedica a la imprenta) desconocido usó una galería de textos y los mezcló de tal manera que logró hacer un libro de textos especimen."
+  },
+  "buy-button": {
+    "text": "Compra la revista"
```
Comment (sssoja) [inline public/locales/es/translation.json:45]: oh good shout

Comment (sssoja) [reply public/locales/es/translation.json:45]: I would say let's check with Aitor if he wants this. I'd say BUY THE MAG is universally understood so might boil down to a design decision

File: src/components/NavMenu.tsx [hunk 2]
```diff
@@ -76,77 +75,44 @@ interface LinkProps {
   url: string;
 }
 
-const Link = ({ page, url }: LinkProps) => {
+const Link = ({ page, url, onClick }: LinkProps & { onClick: () => void }) => {
```
Comment (sssoja) [inline src/components/NavMenu.tsx:78]: nice

File: src/components/BuyButton.tsx [hunk 3]
```diff
@@ -15,79 +14,44 @@ import {
...
-  };
+  const { t } = useTranslation();
 
   return (
     <Fragment>
       {error && <p>{error}</p>}
       <Button
         role="link"
-        onClick={handleClick}
+        onClick={redirectToCheckout(setError)}
```
Comment (sssoja) [inline src/components/BuyButton.tsx:43]: are you getting a typescript error here?

File: src/components/NavMenu.tsx [hunk 4]
```diff
@@ -76,77 +75,44 @@ interface LinkProps {
...
-// };
+const StripeMenuLink: React.FC = () => {
+  const [error, setError] = useState<string>();
+  const { t } = useTranslation();
+
+  return (
+    <Fragment>
+      {error && <p>{error}</p>}
+      <MenuText>
+        <BuyLink onClick={redirectToCheckout(setError)} fontSize={[5, 6, 7, 8]}>
```
Comment (sssoja) [inline src/components/NavMenu.tsx:101]: and here?

File: public/locales/es/translation.json (hunk 1 above)
Comment (sssoja) [reply public/locales/es/translation.json:45]: Yep you're right he wants it

Comment (Jbarget) [reply public/locales/es/translation.json:45]: need to change it to "Comprar mag" i think

File: src/components/BuyButton.tsx (hunk 3 above)
Comment (Jbarget) [reply src/components/BuyButton.tsx:43]: nope its all good for me, sometimes the typescript compiler takes some time to catch up. so closing the file might work or if not, quitting vscode and reopening again should do the trick

image

File: src/components/NavMenu.tsx (hunk 4 above)
Comment (Jbarget) [reply src/components/NavMenu.tsx:101]: nope not for me

**Review by Jbarget — APPROVED** [review]

**Pull request merged**